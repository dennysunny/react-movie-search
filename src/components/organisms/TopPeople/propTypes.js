import { object, array, func } from 'prop-types';

export const propTypes = {
  state: object.isRequired,
  toppeople: array.isRequired,
  handlePageSelect: func.isRequired,
};

export const defaultProps = {
};
