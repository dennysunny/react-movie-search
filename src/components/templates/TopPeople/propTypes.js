import {
  bool,
  object,
  array,
} from 'prop-types';

export const contextTypes = {
  router: object.isRequired,
};

export const propTypes = {
  loading: bool.isRequired,
  toppeople: array.isRequired,
};

export const defaultProps = {
};
