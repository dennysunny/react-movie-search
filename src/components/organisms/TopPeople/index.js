import React from 'react';
import { Link } from 'react-router-dom';

import { propTypes } from './propTypes';
import MyPager from '../../atoms/Pager';
import MediaImage from '../../atoms/MediaImage';

const TopPeopleProfile = ({ toppeople, page, handlePageSelect }) => (
  <div className="top-people">
    <h2>POPULAR PEOPLE</h2>
    <div className="top-people__list">
      {
        toppeople.map(person => (
          <Link key={person.id} to={`/person/${person.id}`}>
            <figure className="top-people__list-element">
              <MediaImage
                mediaType="profile"
                size={1}
                filePath={person.profile_path}
                name={person.name}
                className="top-people__list-element__img"
              />
              <figcaption className="top-people__list-element__name">
                {person.name}
              </figcaption>
            </figure>
          </Link>
        ))
      }
    </div>
    <MyPager
      page={page}
      handlePageSelect={handlePageSelect}
    />
  </div>
);

TopPeopleProfile.propTypes = propTypes;

export default TopPeopleProfile;
