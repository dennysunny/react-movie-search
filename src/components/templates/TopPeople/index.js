import React, { Component } from 'react';

import { propTypes, contextTypes } from './propTypes';
import Spinner from '../../atoms/Spinner';
import PageTransition from '../../atoms/PageTransition/index';
import SEO from '../../atoms/SEO';
import TopPeopleProfile from '../../organisms/TopPeople';

export default class TopPeople extends Component {
  static propTypes = propTypes;

  static contextTypes = contextTypes;

  constructor(props) {
    super(props);

    const params = new URLSearchParams(props.location.search);
    const page = parseInt(params.get('page'), 10);

    this.state = {
      page: page || 1,
    };
  }

  handlePageSelect = (e) => {
    const { router } = this.context;
    if (e > 0) {
      this.setState({ page: e }, () => {
        const { page } = this.state;
        router.history.push(`/person?page=${page}`);
      });
    }
  };

  render() {
    const { loading } = this.props;

    if (loading) {
      return <Spinner />;
    }

    const { toppeople } = this.props;
    const { page } = this.state;
    return (
      <PageTransition>
        <SEO title="Popular people" />
        <div>
          <PageTransition>
            <TopPeopleProfile
              page={page}
              toppeople={toppeople}
              handlePageSelect={this.handlePageSelect}
            />
          </PageTransition>
        </div>
      </PageTransition>
    );
  }
}
