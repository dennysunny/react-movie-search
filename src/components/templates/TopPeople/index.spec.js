import React from 'react';
import { shallow } from 'enzyme';

import TopPeopleProfile from '../../organisms/TopPeople';
import TopPeople from '.';

const mockProps = {
  toppeople: [{
    id: 1,
    name: 'name',
    profile_path: 'path/to/image',
    popularity: 2,
  }],
  location: {
    search: {
      page: 1,
    },
  },
  loading: false,
};

const context = { router: { history: ['/person'] } };

describe('Templates/TopPeople', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<TopPeople {...mockProps} />, { context });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spinner when loading', () => {
    const wrapper = shallow(<TopPeople {...mockProps} loading />, { context });

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle page select', () => {
    const wrapper = shallow(<TopPeople {...mockProps} />, { context });
    wrapper.find(TopPeopleProfile).props().handlePageSelect(2);

    expect(wrapper.state().page).toEqual(2);
  });
});
