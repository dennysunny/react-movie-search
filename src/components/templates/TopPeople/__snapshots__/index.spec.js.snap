// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates/TopPeople should render a spinner when loading 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopPeople
    loading={true}
    location={
      Object {
        "search": Object {
          "page": 1,
        },
      }
    }
    toppeople={
      Array [
        Object {
          "id": 1,
          "name": "name",
          "popularity": 2,
          "profile_path": "path/to/image",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "router": Object {
        "history": Array [
          "/person",
        ],
      },
    },
  },
}
`;

exports[`Templates/TopPeople should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopPeople
    loading={false}
    location={
      Object {
        "search": Object {
          "page": 1,
        },
      }
    }
    toppeople={
      Array [
        Object {
          "id": 1,
          "name": "name",
          "popularity": 2,
          "profile_path": "path/to/image",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <SEO
          title="Popular people"
        />,
        <div>
          <PageTransition>
            <TopPeopleProfile
              handlePageSelect={[Function]}
              state={
                Object {
                  "page": 1,
                }
              }
              toppeople={
                Array [
                  Object {
                    "id": 1,
                    "name": "name",
                    "popularity": 2,
                    "profile_path": "path/to/image",
                  },
                ]
              }
            />
          </PageTransition>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "Popular people",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <PageTransition>
            <TopPeopleProfile
              handlePageSelect={[Function]}
              state={
                Object {
                  "page": 1,
                }
              }
              toppeople={
                Array [
                  Object {
                    "id": 1,
                    "name": "name",
                    "popularity": 2,
                    "profile_path": "path/to/image",
                  },
                ]
              }
            />
          </PageTransition>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <TopPeopleProfile
              handlePageSelect={[Function]}
              state={
                Object {
                  "page": 1,
                }
              }
              toppeople={
                Array [
                  Object {
                    "id": 1,
                    "name": "name",
                    "popularity": 2,
                    "profile_path": "path/to/image",
                  },
                ]
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handlePageSelect": [Function],
              "state": Object {
                "page": 1,
              },
              "toppeople": Array [
                Object {
                  "id": 1,
                  "name": "name",
                  "popularity": 2,
                  "profile_path": "path/to/image",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <SEO
            title="Popular people"
          />,
          <div>
            <PageTransition>
              <TopPeopleProfile
                handlePageSelect={[Function]}
                state={
                  Object {
                    "page": 1,
                  }
                }
                toppeople={
                  Array [
                    Object {
                      "id": 1,
                      "name": "name",
                      "popularity": 2,
                      "profile_path": "path/to/image",
                    },
                  ]
                }
              />
            </PageTransition>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "Popular people",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <PageTransition>
              <TopPeopleProfile
                handlePageSelect={[Function]}
                state={
                  Object {
                    "page": 1,
                  }
                }
                toppeople={
                  Array [
                    Object {
                      "id": 1,
                      "name": "name",
                      "popularity": 2,
                      "profile_path": "path/to/image",
                    },
                  ]
                }
              />
            </PageTransition>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <TopPeopleProfile
                handlePageSelect={[Function]}
                state={
                  Object {
                    "page": 1,
                  }
                }
                toppeople={
                  Array [
                    Object {
                      "id": 1,
                      "name": "name",
                      "popularity": 2,
                      "profile_path": "path/to/image",
                    },
                  ]
                }
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handlePageSelect": [Function],
                "state": Object {
                  "page": 1,
                },
                "toppeople": Array [
                  Object {
                    "id": 1,
                    "name": "name",
                    "popularity": 2,
                    "profile_path": "path/to/image",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "router": Object {
        "history": Array [
          "/person",
        ],
      },
    },
  },
}
`;
